soql-analyzer/src/main/scala/com/socrata/soql/analyzer2/SoQLAnalyzer.scala

```diff
@@ -466,38 +466,63 @@ class SoQLAnalyzer[MT <: MetaTypes] private (
             )
           analyzeStatement(ctx, right, ImplicitFrom.Required(analyzedLeft))
         case other: TrueOp[ast.Select] =>
-          val subCtx =
-            Ctx(
-              scope = ctx.scope,
-              scopedResourceName = None,
-              canonicalName = None,
-              primaryTableName = ctx.primaryTableName,
-              enclosingEnv = ctx.enclosingEnv,
-              udfParams = ctx.udfParams,
-              hiddenColumns = ctx.hiddenColumns,
-              outputColumnHints = ctx.outputColumnHints,
-              attemptToPreserveSystemColumns = false
-            )
+          analyzeTableOp(ctx, other, from0)
+      }
+    }
 
-          val lhsFrom = from0.optionalize(left = true)
-          val lhs = intoStatement(analyzeStatement(subCtx, other.left, lhsFrom))
-          val rhsFrom = from0.optionalize(left = false)
-          val rhs = intoStatement(analyzeStatement(subCtx.copy(primaryTableName = primaryTableName(ctx.scope, other.right)), other.right, rhsFrom))
+    def analyzeTableOp(ctx: Ctx, op: TrueOp[ast.Select], from0: ImplicitFrom): FromStatement = {
+      val subCtx =
+        Ctx(
+          scope = ctx.scope,
+          scopedResourceName = None,
+          canonicalName = None,
+          primaryTableName = ctx.primaryTableName,
+          enclosingEnv = ctx.enclosingEnv,
+          udfParams = ctx.udfParams,
+          hiddenColumns = Set.empty,
```

Comment:
The diff doesn't highlight this change very well, but this line was altered (we're doing hidden-columns as a postprocessing step that removes them now, so as far as subqueries are concerned there are none)